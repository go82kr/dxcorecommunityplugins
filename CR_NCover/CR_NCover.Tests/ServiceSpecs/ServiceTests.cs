using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using CheckPoint;
using ApprovalTests;
using TypeMock.ArrangeActAssert;
using System.Drawing;

namespace CR_NCover.Tests.ServiceSpecs
{
	[TestFixture]
	[Isolated]
	public class ServiceTests
	{
		[Test]
		public void WatchForResultsShouldConfigureWatcher()
		{
			var serviceSentry = Monitor.Interactions(typeof(Service));
			
			Service service = new Service();
			service.WatchForResults("", () => { });

			Approvals.Approve(serviceSentry.Report);
		}

		[Test]
		public void CacheResults()
		{
			var serviceSentry = Monitor.Interactions(typeof(Service));
			
			Isolate.WhenCalled(() => VisualStudio.ActiveFilePath).WillReturn("foo.cs");

			Service service = new Service();
			service.GetCoverageResultsForActiveDocument();
			service.GetCoverageResultsForActiveDocument();

			Approvals.Approve(serviceSentry.Report);
		}
	}
}
