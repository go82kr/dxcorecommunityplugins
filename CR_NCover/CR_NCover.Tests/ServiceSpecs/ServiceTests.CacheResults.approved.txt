(System.IO.FileSystemWatcher) watcher..cctor()
(System.IO.FileSystemWatcher) watcher..ctor()
